### PR TITLE
[DM-35171] Add HiPS bucket for DP0.2

### PR DESCRIPTION
**Pull request by rra**
Use the naming convention discussed on Slack.  Grant RW access to
the Butler service account we've been using for ease of importing
data (the plan is to remove that access once the data is imported).

Create the crawlspace service account for data-int and add it to
the HiPS ACL list.

### CODE REVIEW

File: environment/deployments/data-curation/main.tf [hunk 1]
```diff
@@ -115,7 +115,7 @@ module "storage_bucket_2" {
   }
 }
 
-// HIPS Storage Bucket
+// HIPS Storage Bucket (VISA testing)
```
Comment (gpdf) [inline environment/deployments/data-curation/main.tf:118]: Was that "VISA" supposed to be "VISTA"?

Comment (ktlim) [inline environment/deployments/data-curation/main.tf]:
```suggestion
// HIPS Storage Bucket (VISTA testing)
```

File: environment/deployments/data-curation/main.tf [hunk 2]
```diff
@@ -138,9 +138,33 @@ module "storage_bucket_3" {
...
+  members = var.hips_service_accounts
+}
+
+// HIPS Storage Bucket (DP0.2)
+module "storage_bucket_4" {
+  source        = "../../../modules/bucket"
+  project_id    = module.project_factory.project_id
+  storage_class = "REGIONAL"
+  location      = "us-central1"
+  suffix_name   = ["dp02-hips"]
```
Comment (gpdf) [inline environment/deployments/data-curation/main.tf:150]: Appears to conform to the naming convention we agreed on.

File: environment/deployments/data-curation/main.tf (hunk 1 above)
Comment (gpdf) [inline environment/deployments/data-curation/main.tf:118]: Spelling is "HiPS" - might as well get it right now.

File: environment/deployments/data-curation/main.tf [hunk 3]
```diff
@@ -138,9 +138,33 @@ module "storage_bucket_3" {
 resource "google_storage_bucket_iam_binding" "hips-vista-bucket-ro-iam-binding" {
   bucket  = module.storage_bucket_3.name
   role    = "roles/storage.objectViewer"
-  members = [
-    "serviceAccount:${var.hips_service_account}"
-  ]
+  members = var.hips_service_accounts
+}
+
+// HIPS Storage Bucket (DP0.2)
```
Comment (gpdf) [inline environment/deployments/data-curation/main.tf:144]: Spelling is "HiPS" - might as well get it right now.

**Review by gpdf — COMMENTED** [review]
OK insofar as I'm able to review it.  No comment on any of the technicalities other than the naming convention.

File: environment/deployments/data-curation/main.tf (hunk 1 above)
Comment (rra) [reply environment/deployments/data-curation/main.tf]: Yup, thanks for that and spotting the other typos. Fixed now.

File: environment/deployments/data-curation/main.tf [hunk 4]
```diff
@@ -138,9 +138,33 @@ module "storage_bucket_3" {
...
+  prefix_name   = "static"
+  versioning = {
+    dp02-hips = false
+  }
+  force_destroy = {
+    dp02-hips = false
+  }
+  labels = {
+    environment = var.environment
+    application = "hips"
```
Comment (gpdf) [inline environment/deployments/data-curation/main.tf:160]: What does this do?

Comment (ktlim) [reply environment/deployments/data-curation/main.tf:160]: It shows up in the browser:
Screen Shot 2022-06-09 at 16 32 43

**Review by ktlim — COMMENTED** [review]
Overall, LGTM but I'm not a TF expert.

**Review by dspeck1 — APPROVED** [review]
looks good

**Pull request merged**